### PR TITLE
Dependency Updates

### PR DESCRIPTION
**Pull request by kingthorin**
- build.gradle.kts > Update dependency checker dependency ;)
- LEGALNOTICE.md > Document dependency changes.
- zap.gradle.kts > Update various dependencies.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
jgrapht-core could go from 0.9.0 to 1.5.1 but there seems to be breaking changes. I didn't have time to address this in the core and wasn't sure if it might break add-ons.

JUnit/Jupiter and Gradle could also be upgraded but both the "latests" seem to be on release candidates or pre-release.
```console
 - org.junit.jupiter:junit-jupiter-api [5.7.0 -> 5.8.0-M1]
     https://junit.org/junit5/
 - org.junit.jupiter:junit-jupiter-engine [5.7.0 -> 5.8.0-M1]
     https://junit.org/junit5/
 - org.junit.jupiter:junit-jupiter-params [5.7.0 -> 5.8.0-M1]
     https://junit.org/junit5/
 - org.mockito:mockito-junit-jupiter [3.6.28 -> 3.8.0]
     https://github.com/mockito/mockito

Gradle release-candidate updates:
 - Gradle: [5.4.1 -> 6.8.3 -> 7.0-rc-2]
```

I was confused about the releases for Ice4j. Maven does seem to have the artifact mentioned, but the GitHub repo only goes up to 3.0.0, and the trailing string in the suggestion seemed out of place to me.
```console
 - org.jitsi:ice4j [1.0 -> 3.0-24-g34c2ce5]
     https://github.com/jitsi/ice4j
```

Lastly apparently laungh4j could be upgraded but I couldn't see where this dependency is defined (perhaps this is Gradle or Eclipse packaged?):
```console
 - net.sf.launch4j:launch4j [3.12 -> 3.14]
     http://sourceforge.net/projects/launch4j/
```

File: LEGALNOTICE.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ and subject to their respective licenses.
 | commons-io-2.8.0.jar                | Apache 2.0                |
 | commons-jxpath-1.3.jar              | Apache 2.0                |
 | commons-lang-2.6.jar                | Apache 2.0                |
-| commons-lang3-3.11.jar              | Apache 2.0                |
+| commons-lang3-3.12.0.jar              | Apache 2.0                |
```
Comment (thc202) [inline LEGALNOTICE.md:47]: Remove one space.

File: LEGALNOTICE.md [hunk 2]
```diff
@@ -58,14 +58,14 @@ and subject to their respective licenses.
...
 | json-lib-2.4-jdk15.jar              | MIT + "Good, Not Evil"    |
-| log4j-1.2-api-2.14.0.jar            | Apache 2.0                |
-| log4j-api-2.14.0.jar                | Apache 2.0                |
-| log4j-core-2.14.0.jar               | Apache 2.0                |
+| log4j-1.2-api-2.14.1.jar            | Apache 2.0                |
+| log4j-api-2.14.1.jar                | Apache 2.0                |
+| log4j-core-2.14.1.jar               | Apache 2.0                |
 | rsyntaxtextarea-3.1.2.jar           | BSD-3 clause              |
-| sqlite-jdbc-3.32.3.2.jar            | BSD-2 clause              |
+| sqlite-jdbc-3.34.0.jar            | BSD-2 clause              |
```
Comment (thc202) [inline LEGALNOTICE.md:68]: Add one space.

**Comment by thc202** [comment]
Dependencies declared as API should not be updated between major versions, as they usually have breaking changes (which could break add-ons), unless they can coexist in which case we would have to bundle both.

The ice4j is in Maven Central: https://search.maven.org/artifact/org.jitsi/ice4j/3.0-24-g34c2ce5/jar
The version uses the build number(?) and the commit hash (which matches to what is in the repo).

The launch4j is a transitive dependency, it will be updated along with the Gradle plugin (once released).

**Comment by kingthorin** [comment]
Done & done

**Comment by thc202** [comment]
No intention to update ice4j?

**Comment by kingthorin** [comment]
Sorry I hadn't checked if it fell into that api/major version category but it seems it doesn't. Tweaked now as well.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**